server/service/software_titles.go

```diff
@@ -177,6 +177,9 @@ func (svc *Service) SoftwareTitleByID(ctx context.Context, id uint, teamID *uint
 
 			return nil, fleet.NewPermissionError("Error: You don't have permission to view specified software. It is installed on hosts that belong to team you don't have permissions to view.")
 		}
+		if fleet.IsNotFound(err) {
```

Comment:
Feels like we shouldn't need this, given that we're using the fame fn call in `server/datastore/mysql/hosts.go` L1104, which means that that call would work just as well when wrapped. Or am I missing something here?